### PR TITLE
task: enable thelper and errorlint

### PR DESCRIPTION
**Pull request by gssbzn**
## Description

Enables `thelper`, this linter makes sure that test helper functions call `t.Helper`

Enables `errorlint`, making sure error handling follow go1.13 error handling best practices

### CODE REVIEW

File: mongodbatlas/mongodbatlas.go [hunk 1]
```diff
@@ -169,24 +169,6 @@ type ListOptions struct {
 	ItemsPerPage int `url:"itemsPerPage,omitempty"`
 }
 
-// ErrorResponse reports the error caused by an API request.
-type ErrorResponse struct {
```
Comment (gssbzn) [inline mongodbatlas/mongodbatlas.go:173]: moved closer to the implementation of `Error`

File: mongodbatlas/mongodbatlas.go [hunk 2]
```diff
@@ -222,7 +204,7 @@ func (resp *Response) CurrentPage() (int, error) {
 
 	pageNum, err := strconv.Atoi(pageNumStr)
 	if err != nil {
-		return 0, fmt.Errorf("error getting current page: %s", err)
+		return 0, fmt.Errorf("error getting current page: %w", err)
```
Comment (gssbzn) [inline mongodbatlas/mongodbatlas.go:207]: `w` is a go1.13 wrapper for errors which allows `error.Is` to map wrapped errors

File: .golangci.yml [hunk 3]
```diff
@@ -56,54 +60,79 @@ linters:
...
+    - depguard # Go linter that checks if package imports are in a list of acceptable packages [fast: false, auto-fix: false]
+    - dogsled # Checks assignments with too many blank identifiers (e.g. x, _, _, _, := f()) [fast: true, auto-fix: false]
+    - errcheck # Errcheck is a program for checking for unchecked errors in go programs. These unchecked errors can be critical bugs in some cases [fast: false, auto-fix: false]
+    - errorlint # errorlint is a linter for that can be used to find code that will cause problems with the error wrapping scheme introduced in Go 1.13. [fast: false, auto-fix: false]
+    - exhaustive # check exhaustiveness of enum switch statements [fast: false, auto-fix: false]
+    - exportloopref # checks for pointers to enclosing loop variables [fast: false, auto-fix: false]
+    - gochecknoinits # Checks that no init functions are present in Go code [fast: true, auto-fix: false]
+    - gocritic # Provides many diagnostics that check for bugs, performance and style issues. [fast: false, auto-fix: false]
+    - gocyclo # Computes and checks the cyclomatic complexity of functions [fast: true, auto-fix: false]
```
Comment (threebee) [inline .golangci.yml:73]: Thank you for adding the description for each linter!

**Review by threebee — APPROVED** [review]
LGTM!

**Pull request merged**